mesonbuild/arglist.py

```diff
@@ -267,12 +269,15 @@ def extend_preserving_lflags(self, iterable: T.Iterable[str]) -> None:
         normal_flags = []
         lflags = []
         for i in iterable:
-            if i not in self.always_dedup_args and (i.startswith('-l') or i.startswith('-L')):
+            if i not in self.always_dedup_args and (i.startswith('-l') or i.startswith('-L')) and i not in self._libs:
```

Comment:
I'm never sure about `-lfoo` order, shouldn't we keep last occurrence instead of first?

Reply:
I'm really not sure myself. Probably it should keep the last occurence, because as libs are visited, symbols to be linked are accumulated. But in that case, -L order may become wrong... And there may be linkers that requires to have some libs duplicated for static linking, in case of circular references.